### PR TITLE
Document nonce handling and contract state tree updates.

### PR DESCRIPTION
**Pull request by happi**

### CODE REVIEW

File: contracts/contract_state_tree.md [hunk 1]
```diff
@@ -13,24 +13,50 @@ The contract state tree contains objects:
...
 - The contract log (data binary)
 - Active flag (Boolean)
-- List of refering contracts
+- List of referring contracts (Note this will be changed to a list of referred contracts and a refcount.)
 - Deposit (amount)
 - vm_version
 
 See [Contract Serialization](../serializations.md#contract) for the serialization specification.
 
+The contract also has a nonce and a balance recorded in an account state tree.
```
Comment (lucafavatella) [inline contracts/contract_state_tree.md]: I see this refers to https://github.com/aeternity/epoch/pull/1131 ok

File: contracts/contract_state_tree.md [hunk 2]
```diff
@@ -13,24 +13,50 @@ The contract state tree contains objects:
...
 
+TO BE WRITTEN.
+
 ## Calls state tree
 
-The calls state tree contains all calls done in the block
-and their return values.
+The calls state tree contains all calls done in the block and their
+return values. The calls state tree is pruned in each block so that
+the lates block only contains the calls in that block.
```
Comment (lucafavatella) [inline contracts/contract_state_tree.md]: lates?

File: contracts/contract_state_tree.md [hunk 3]
```diff
@@ -13,24 +13,50 @@ The contract state tree contains objects:
...
+  etc.) the nonce of the contract is increased by one.
+  (This is subject to change in future versions.)
+
+The contract's state tree is updated:
+- Upon success of a contract call the contract log and the contract's storage sub tree is updated.
+- When another contract is created and that new contract referrers to the contract in the code, the refcount is increased.
+- When another contract that refers to the contract is deleted the refcount is decreased.
+- When the contract executes the deactivate instruction the active field is set to false.
+
+The fields owner, vm_version, code, and deposit are static throughout a contracts life time.
```
Comment (lucafavatella) [inline contracts/contract_state_tree.md]: contracts -> contract's

File: contracts/contract_state_tree.md [hunk 4]
```diff
@@ -13,24 +13,50 @@ The contract state tree contains objects:
...
+  - Through a call to the contract, with the transferred amount of the call.
+  - Through a spend from a known account with an attached contract.
+- When a contract's balance is updated by a call executing a spend transaction.
+- When a call to a contract executes a transaction (create contract, call contract, spend,
+  etc.) the nonce of the contract is increased by one.
+  (This is subject to change in future versions.)
+
+The contract's state tree is updated:
+- Upon success of a contract call the contract log and the contract's storage sub tree is updated.
+- When another contract is created and that new contract referrers to the contract in the code, the refcount is increased.
```
Comment (lucafavatella) [inline contracts/contract_state_tree.md]: referrers -> refers?

**Comment by happi** [comment]
@lucafavatella The key generation is documented in https://github.com/aeternity/protocol/blob/master/contracts/contract_transactions.md#create-contract-transaction

**Pull request merged**